test/e2e/networkpolicy_test.go

```diff
@@ -30,6 +33,108 @@ import (
+		antreaControllerConf, _ := data["antrea-controller.conf"]
+		antreaControllerConf = strings.Replace(antreaControllerConf, "#  NetworkPolicyMetrics: false", "  NetworkPolicyMetrics: true", 1)
+		data["antrea-controller.conf"] = antreaControllerConf
+		antreaAgentConf, _ := data["antrea-agent.conf"]
+		antreaAgentConf = strings.Replace(antreaAgentConf, "#  NetworkPolicyMetrics: false", "  NetworkPolicyMetrics: true", 1)
+		data["antrea-agent.conf"] = antreaAgentConf
+	}, true, true); err != nil {
+		t.Fatalf("Failed to enable NetworkPolicyMetrics feature: %v", err)
+	}
+
+	serverName, serverIP, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, "test-server-", "")
+	defer cleanupFunc()
+
+	clientName, _, cleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", "")
+	defer cleanupFunc()
+
+	np1, err := data.createNetworkPolicy("test-networkpolicy-ingress", &networkingv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		Ingress: []networkingv1.NetworkPolicyIngressRule{{
+			From: []networkingv1.NetworkPolicyPeer{{
+				PodSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"antrea-e2e": clientName,
+					},
+				}},
+			},
+		}},
+	})
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(np1); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+	np2, err := data.createNetworkPolicy("test-networkpolicy-egress", &networkingv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+		Egress: []networkingv1.NetworkPolicyEgressRule{{
+			To: []networkingv1.NetworkPolicyPeer{{
+				PodSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"antrea-e2e": serverName,
+					},
+				}},
+			},
+		}},
+	})
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(np2); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	// Wait for a few seconds in case that connections are established before policies are enforced.
+	time.Sleep(2 * time.Second)
+
+	sessions := 10
+	var wg sync.WaitGroup
+	for i := 0; i < sessions; i++ {
+		wg.Add(1)
+		go func() {
+			cmd := []string{"/bin/sh", "-c", fmt.Sprintf("nc -vz -w 4 %s 80", serverIP)}
+			data.runCommandFromPod(testNamespace, clientName, busyboxContainerName, cmd)
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	if err := wait.Poll(5*time.Second, defaultTimeout, func() (bool, error) {
+		for _, np := range []string{"test-networkpolicy-ingress", "test-networkpolicy-egress"} {
+			metric, err := data.crdClient.MetricsV1alpha1().NetworkPolicyMetricses(testNamespace).Get(context.TODO(), np, metav1.GetOptions{})
```

Comment:
ewww didn't realize we would end up with `NetworkPolicyMetricses` :P

Reply:
I just took a look at how the code is generated and found a way to add exception, otherwise it simply adds "es" if the last letter is "s", updated in latest patch,  see change made to client-gen command.